gst/isomp4/qtdemux.c

```diff
@@ -6341,7 +6341,7 @@ qtdemux_parse_samples (GstQTDemux * qtdemux, QtDemuxStream * stream, guint32 n)
         GST_LOG_OBJECT (qtdemux, "Created entry %d with offset "
             "%" G_GUINT64_FORMAT, j, cur->offset);
 
-        if (stream->samples_per_frame * stream->bytes_per_frame) {
+        if (stream->samples_per_frame && stream->bytes_per_frame) {
           cur->size =
```

Comment:
It's still being a int in boolean context, for me it's more clear:

if ((stream->samples_per_frame !=0 ) && ( stream->bytes_per_frame != 0)) 

But this is stylish appreciation, not a PR failure. Approved.

Reply:
Well, I'm actually surprised the warning shows up with the `*` and not now with the `&&`, since the `!= 0` is what actually avoids the implicit cast from int to bool. Unless the compiler only checks single statements for that warning, that's it :smile:

Reply:
Probably because the implicit cast can't fail in this case, the machine code with` !=` and without  `!=` is probably the same.
It's only a matter of stylish (for me it's more readable the explicit `!=` conversion, but it's only a personal appreciation).

In the case of the multiplication however, the boolean may fail: exist a possibility of having two non-zero numbers that result in zero due to a integer overflow. 

The compiler warns with a good reason because this is not probably what you want.
On the other hand, using * instead && is a code obfuscation that makes baby jesus cry.

Reply:
>In the case of the multiplication however, the boolean may fail: exist a possibility of having two non-zero numbers that result in zero due to a integer overflow.

What? You just blew my mind. My understanding is that signed interger overflow is undefined behavior and anything may happen afterwards. In this case, however, both variables are defined as `guint`. I thought the result of the multiplication would be stored in a `guint` as well (not promoted to int), where the overflow is defined behavior and can never be 0 unless one of the values is 0

Reply:
It is undefined behaviour because the language specification does not define what happens in case of an overflow.
This is, it is implementation specific, not language-specific. 
For two integers in x86 architecture, it's perfectly possible to pass that IF with an overflow.

`#include <stdio.h>

int main(int argc, const char *argv[]) {

  unsigned int a = 0x10000000;
  unsigned int b = 0x10;

  printf("result: %d\n\n", a * b);

  if (a * b) {
    printf(":-)");
  } else {
    printf("whaaat??\n");
  }

  if (a && b) {
    printf("- Both are non-zero");
  }
}
`

result: 
```
result: 0

whaaat??
- Both are non-zero
```